### DISCUSSION TITLE
Using custom live data source and custom broker.

### DISCUSSION THREAD

**Discussion by rohit1998** [nautechsystems/nautilus_trader#572] [Q&A] [answered]
I have a framework where I can get my own custom live data feed and use a custom trade api from my broker. I need a backtesting+live trade framework to code strategies etc. I also plan to deploy using same. I see this has support for live trading. Is there functionality to add a generic data feed class where i can implement how to get data and it handles how to get data to trade engine. And then at time of execution i can define how to place order.

**Comment by cjdsellers** [comment, marked as answer]
Apologies for the very delayed response!

The short answer is, absolutely the platform can handle your use case. You can implement a custom `LiveDataClient` for your feed https://github.com/nautechsystems/nautilus_trader/blob/develop/nautilus_trader/adapters/_template/data.py.

You can also backtest on whatever sort of `GenericData` this component produces.

Then simply write a `TradingStrategy`.
https://docs.nautilustrader.io/user_guide/strategies.html
